Revision of the content structure (#727)
italia/eidas-it-wallet-docs: docs/en/how-to-read-spec.rst

**Section** :ref:`endpoints:Endpoints`: 
  Technical specifications for all API endpoints exposed by each entity type, including federation endpoints and specialized service integrations.

**Section** :ref:`algorithms:Cryptographic Algorithms`, :ref:`security-privacy-considerations:Security and Privacy Considerations`, and :ref:`log-retention-policy:General Log Retention Policies` (**Implementation Support**): 
  Cryptographic requirements, security and privacy considerations, and log retention policies essential for compliant implementations.

**Section** :ref:`defined-terms-and-references:Defined Terms and References`, :ref:`contribute:How to contribute`, and :ref:`open-source:Open Source Releases` (**Terminology and References**):
  Comprehensive terminology, normative references, contribution guidelines.

**Section** :ref:`appendix:Appendix`: 
  Provides supplementary technical details, implementation patterns, and testing frameworks including mobile application instance management, national platform integration specifications, and comprehensive test matrices for ecosystem validation.

Reading Paths by Role
---------------------

> Before diving into role-specific sections, all readers should be familiar with the foundational concepts outlined in Sections :ref:`introduction:Introduction`, :ref:`architecture-overview:Architecture Overview`, and :ref:`trust:The Infrastructure of Trust`, which establish the common vocabulary, design principles, and trust infrastructure that provide the underlying framework for the entire ecosystem.

Reviewer: If I were the King I would say "MUST be familiar with ..."

Wallet Provider
^^^^^^^^^^^^^^^^^^^^

> Readers implementing or operating a **Wallet Provider** Solution should focus on understanding how to securely manage Digital Credentials and facilitate user interactions with other ecosystem participants.

Reviewer: ```suggestion
Readers implementing or operating a **Wallet Provider** Solution should focus on understanding how to securely manage Digital Credentials and facilitate User interactions with other ecosystem participants.
```

* **Section** :ref:`digital-credential-flows:Digital Credential Flows`: Issuance and presentation flows for Digital Credentials.
* **Section** :ref:`wallet-provider-endpoint:Wallet Provider Endpoints`: Federation and IT-Wallet specific API specifications.
* **Section** :ref:`algorithms:Cryptographic Algorithms`: Security and cryptographic implementation requirements.
* **Section** :ref:`mobile-application-instance:Mobile Application Instance`: Mobile-specific implementation patterns.
* **Section** :ref:`e-service-pdnd:e-Service PDND`: Integration with the National Data Interoperability Platform.

**Secondary sections:**

* **Section** :ref:`credential-issuer-solution:Credential Issuer Solution`: Understanding issuer interactions and requirements.
* **Section** :ref:`relying-party-solution:Relying Party Solution`: Understanding RP interactions and presentation protocols.

Credential Issuer
^^^^^^^^^^^^^^^^^^^^

For readers who are interested in implementing a **Credential Issuer** Solution, they should focus on the Digital Credential lifecycle and issuance interaction flows.

**Essential sections:**

> * **Section** :ref:`credential-issuer-solution:Credential Issuer Solution`: Credential Issuer Solution - Complete issuer implementation requirements and component details.

Reviewer: ```suggestion
* **Section** :ref:`credential-issuer-solution:Credential Issuer Solution`: Credential Issuer Solution - Complete Issuer implementation requirements and component details.
```

* **Section** :ref:`authentic-sources:Authentic Sources`: Understanding authoritative data source integration patterns.
* **Section** :ref:`digital-credential-management:Digital Credential Management`: Digital Credential formats and lifecycle management.
* **Section** :ref:`credential-issuance:Digital Credential Issuance`: Detailed issuance flow implementation.
* **Section** :ref:`credential-issuer-endpoint:Credential Issuer Endpoints`: Federation and issuance-specific API specifications.
* **Section** :ref:`algorithms:Cryptographic Algorithms`:  Signing algorithms and security implementation requirements.

**Secondary sections:**

* **Section** :ref:`wallet-solution:Wallet Solution`: Understanding Wallet Instance interactions and requirements.
* **Section** :ref:`credential-presentation:Digital Credential Presentation`: Understanding how Digital Credentials are presented in both remote and proximity scenario.
* **Section** :ref:`e-service-pdnd:e-Service PDND`: Integration with the National Data Interoperability Platform.

.. note::

>     If the Credential Issuer must authenticate the User, Section :ref:`credential-presentation:Digital Credential Presentation` is essential. If the Authentic Source providing User's attributes belongs to the public sector, Section :ref:`e-service-pdnd:e-Service PDND` is also essential.  

Reviewer: ```suggestion
    If the Credential Issuer authenticates the User it must comply with Section :ref:`credential-presentation:Digital Credential Presentation`. If the Authentic Source providing User's attributes belongs to the public sector it must comply with Section :ref:`e-service-pdnd:e-Service PDND`.  
```

**Essential sections:**

* **Section** :ref:`relying-party-solution:Relying Party Solution`: Complete verifier implementation requirements and Entity Configuration.
* **Section** :ref:`digital-credential-management:Digital Credential Management`: Understanding Digital Credential formats and validity verification.
* **Section** :ref:`credential-presentation:Digital Credential Presentation`: Presentation flow implementation for both remote and proximity scenarios.
* **Section** :ref:`relying-party-endpoint:Relying Party Endpoints`: Federation and verification related API specifications.
* **Section** :ref:`algorithms:Cryptographic Algorithms`: Cryptographic suite requirements.

**Secondary sections:**

* **Section** :ref:`wallet-solution:Wallet Solution`: Understanding Wallet Instance interactions and presentation protocols
* **Section** :ref:`mobile-application-instance:Mobile Application Instance`: Mobile-specific implementation patterns.

Cross-Cutting Topics
----------------------

Regardless of your primary role, certain sections contain information relevant to all ecosystem participants:

**Security and Compliance:**

>     All implementers must thoroughly review Section :ref:`security-privacy-considerations:Security and Privacy Considerations` and Section :ref:`log-retention-policy:General Log Retention Policies` to ensure proper security posture, privacy protection, and regulatory compliance within the IT-Wallet ecosystem.

Reviewer: ```suggestion
    All implementers must implement their solutions according to Section :ref:`security-privacy-considerations:Security and Privacy Considerations` and Section :ref:`log-retention-policy:General Log Retention Policies`.
```

**Standards and References:**

>     Section :ref:`defined-terms-and-references:Defined Terms and References` provides essential normative references, defined terms, and technical standards that establish the foundation for interoperable implementations across all ecosystem participants.

Reviewer: ```suggestion
    Section :ref:`defined-terms-and-references:Defined Terms and References` provides normative references, defined terms, and technical standards to enable secure and proper interoperability among all participants.
```

**Testing and Validation:**

>     Section :ref:`test-plans:Test Plans` provides a comprehensive test matrix for validating implementations across different roles and interaction flows, ensuring compliance with specification requirements and interoperability standards.

Reviewer: ```suggestion
    Section :ref:`test-plans:Test Plans` provides a comprehensive test matrix for validating implementations across different roles and interaction flows.
```

Implementation Approach
-----------------------

The following phased reading approach is suggested:

    1. **Foundation Phase**: Read  Sections :ref:`introduction:Introduction`, :ref:`architecture-overview:Architecture Overview`, and :ref:`trust:The Infrastructure of Trust` to establish conceptual understanding of the IT-Wallet paradigm, design principles, and trust infrastructure.
    2. **Role-Specific Phase**: Focus on primary role's essential sections to understand specific implementation requirements, main technical component, the general architecture and interaction flows (see Section :ref:`entities:Entities` and Section :ref:`endpoints:Endpoints` for more details).
    3. **Integration Phase**: Review secondary sections relevant to interactions with other ecosystem participants and platform integration requirements.

>     4. **Validation Phase**: Study security considerations, testing guidance, and compliance requirements to ensure robust and compliant implementations (see Sections :ref:`security-privacy-considerations:Security and Privacy Considerations`, :ref:`log-retention-policy:General Log Retention Policies`, and :ref:`test-plans:Test Plans` for additional information).

Reviewer: ```suggestion
    4. **Validation Phase**: Study security considerations, testing guidance, and compliance requirements according to Sections :ref:`security-privacy-considerations:Security and Privacy Considerations`, :ref:`log-retention-policy:General Log Retention Policies`, and :ref:`test-plans:Test Plans` for additional information.
```

.. note::

>     For implementers working on solutions that span multiple roles (e.g., a combined Issuer Relying Party Solutions), it is recommended reviewing the essential sections for all relevant roles before proceeding to implementation. It is important to take particular note of Entity Configuration requirements and federation flows that apply to multiple roles.

Reviewer: ```suggestion
    For implementers working on solutions that span multiple roles (e.g., a combined Issuer Relying Party Solutions), it is recommended reviewing the sections for all relevant roles before proceeding with the developments. It is important to take particular note of Entity Configuration requirements and federation flows that apply to multiple roles.
```
